lib/inferno/app/endpoint.rb

```diff
@@ -53,23 +57,32 @@ class Endpoint < Sinatra::Base
       helpers Helpers::Configuration
       helpers Helpers::BrowserLogic
 
-      set :public_folder, (proc { File.join(root, '../../public') })
+      set :public_folder, (proc { File.join(root, '../../../public') })
       set :static, true
       set :views, File.expand_path('views', __dir__)
       set(:prefix) { '/' << name[/[^:]+$/].underscore }
 
+      @@inferno_modules = settings.modules.map { |m| Inferno::Module.get(m) }.compact
+
+      def self.inferno_modules
+        @@inferno_modules
+      end
+
+      def self.inferno_modules=(new_modules)
+        @@inferno_modules = new_modules
+      end
+
       def render_index
-        unless defined?(settings.presets).nil? || settings.presets.nil?
+        unless @@inferno_modules
```

Comment:
What does this change do?

Reply:
There was no way to change the modules available in the UI except through the config file.  This change, and few methods above it, allowed the modules to be changed programmatically with

```ruby
    # Add to module list as the only module (can << if you want to keep Inferno defaults)
    example_module = Inferno::Module.get('example_read')
    Inferno::App::Endpoint.inferno_modules = [ example_module ]
    Inferno::App.new
```

```ruby
Inferno::App::Endpoint.inferno_modules << example_module
``` 
also works if you want to append on what is already there (i.e. what is in the default inferno config)